### DISCUSSION TITLE
Excel-DNA 1.6.0

### DISCUSSION THREAD

**Discussion by hxhgxy** [Excel-DNA/ExcelDna#624] [Q&A] [answered]
How can I make CTP in .Net 6.0 project by using Excel-DNA 1.6.0?

**Comment by govert** [comment, marked as answer]
You need to fiddle with the clas declaration, to add an explicit interface. See this discussion https://groups.google.com/g/exceldna/c/onRDZLJwoBY/m/ddsjBDbsBAAJ

**Comment by hxhgxy** [comment]
thanks a lot!

In Net framework 4.8 project, i can use wpf usercontrol onto CTP. Is there any solution to use WPF controls on CTP?
Seems no library support both winform and WPF, except for Net framework, right?

**Comment by govert** [comment]
WPF works the same under .NET 6 as under .NET Framework. You need a host control from WinForms to put the WPF into a CTP.

**Comment by hxhgxy** [comment]
I know ElementHost in WinForm. However, I cannot create a WPF UserControl in WinForm Library project as in Net Framework project. 
If I create a WPFLibrary project, I don't know where to find WinForm usercontrol for CTP. Which kind of Library project I should to use? What kind of references I should to add?
Thanks a lot!

**Comment by hxhgxy** [comment]
After putting `<UseWpf>true</UseWpf>` in csproj file, I can add WPF usercontrol to project of WInform Library.

**Comment by digiupdatehub** [comment]
HOW TO BE AN EXPERT IN EXCEL
https://www.digiupdatehub.tech/2023/07/how-to-be-expert-in-excel.html?m=1
